src/components/tableV2/core/base-table.js

```diff
@@ -255,39 +253,41 @@ Table.Body = forwardRef(({ children, ...props }, ref) => (
 Table.Cell = forwardRef(
   (
     {
+      align = "left",
+      cellStyles = {},
       children,
-      onClick,
-      width,
       maxWidth,
       minWidth,
-      align = "left",
-      cellStyles = {},
+      onClick,
+      pinnedStyles,
       styles = {},
-      ...props
+      width,
+      ...rest
     },
     ref
   ) => {
     const handleClick = e => {
       e.persist()
-      if (props.stopPropagation) e.stopPropagation()
+      if (rest.stopPropagation) e.stopPropagation()
       onClick?.()
     }
     return (
       <Box
-        width={{ max: `${maxWidth}px`, base: `${width}px`, min: `${minWidth}px` }}
+        as="td"
+        onClick={handleClick}
         padding={[3]}
+        ref={ref}
         sx={{
           textAlign: align,
           height: "65px",
           maxHeight: "65px",
           whiteSpace: "nowrap",
           ...cellStyles,
+          ...(pinnedStyles || {}),
```

Comment:
consistent with present approach would be to have `pinnedStyles = {}` when destructuring (just as `styles`)